app/src/main/kotlin/com/wire/android/ui/home/settings/backup/BackUpAndRestoreStateHolder.kt

```diff
@@ -0,0 +1,41 @@
+package com.wire.android.ui.home.settings.backup
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+
+class BackUpAndRestoreStateHolder {
+
+    var dialogState: BackupAndRestoreDialog by mutableStateOf(
+        BackupAndRestoreDialog.None
+    )
+
+    fun showBackupDialog() {
+        dialogState = BackupAndRestoreDialog.Backup
+    }
+
+    fun showRestoreDialog() {
+        dialogState = BackupAndRestoreDialog.Restore
+    }
+
+    fun dismissDialog() {
+        dialogState = BackupAndRestoreDialog.None
+    }
+
+}
+
+@Composable
+fun rememberBackUpAndRestoreStateHolder(): BackUpAndRestoreStateHolder {
+    return remember {
+        BackUpAndRestoreStateHolder()
+    }
+}
+
+sealed class BackupAndRestoreDialog {
+    object None : BackupAndRestoreDialog()
+    object Backup : BackupAndRestoreDialog()
+    object Restore : BackupAndRestoreDialog()
```

Comment:
Not sure if these values imply current state of the backup process or simply type of dialog state. I guess it's the latter one, so for more clarity can we rename these to:
```suggestion
    object CreateBackup : BackupAndRestoreDialog()
    object RestoreBackup : BackupAndRestoreDialog()
```